gwctl/pkg/resourcediscovery/discoverer.go

```diff
@@ -444,6 +534,47 @@ func fetchHTTPRoutes(ctx context.Context, k8sClients *common.K8sClients, filter
 	return httpRouteList.Items, nil
 }
 
+// fetchHTTPRoutes fetches HTTPRoutes based on a filter.
+func (d Discoverer) fetchReferenceGrants(ctx context.Context, filter Filter) ([]gatewayv1beta1.ReferenceGrant, error) {
+	gvr := schema.GroupVersionResource{
+		Group:    defaultReferenceGrantGroupVersion.Group,
+		Version:  defaultReferenceGrantGroupVersion.Version,
+		Resource: "referencegrants",
+	}
+	// TODO(gauravkghildiyal): Uncomment once
+	// https://github.com/kubernetes-sigs/gateway-api/pull/3001 merges
+	// if d.PreferredReferenceGrantGroupVersion != (metav1.GroupVersion{}) {
+	// 	gvr.Version = d.PreferredReferenceGrantGroupVersion.Version
+	// }
```

Comment:
Reminder that https://github.com/kubernetes-sigs/gateway-api/pull/3001 has merged

Reply:
Thanks for the reminder!